core/parser/src/parser/statement/declaration/hoistable/class_decl/mod.rs

```diff
@@ -89,6 +89,10 @@ where
                 BindingIdentifier::new(self.allow_yield, self.allow_await)
                     .parse(cursor, interner)?
             }
+            TokenKind::Keyword((k, _)) if !k.is_reserved_keyword() => {
+                BindingIdentifier::new(self.allow_yield, self.allow_await)
+                    .parse(cursor, interner)?
+            }
```

Comment:
Suggestion: Consider adding a test for this

Reply:
Yes, it looks good, thanks for adding the test! I suggested to add a rust test because the [test262 test suite](https://github.com/tc39/test262) didn't cover it.